### PR TITLE
Fixes Setup.py error on Windows #676

### PR DESCRIPTION
**Pull request by Sai-Adarsh**
Fixes Setup.py error on Windows, added shell script to check python environment, prints the python version, copy files from /kubernetes/base to /kubernetes/ and finally runs python setup.py install issue #676

### CODE REVIEW

**Comment by Sai-Adarsh** [comment]
@micw523 gave a PR. could you check this out.

File: scripts/setup-fix.bat [hunk 1]
```diff
@@ -0,0 +1,11 @@
+if ! [ -x "$(command -v python)" ];
+then 
+	echo Python environment notfound..
+else 
+	echo Python environment found..
+fi
+python --version
+cp -R ../kubernetes/base/* ../kubernetes/
```
Comment (yliaog) [inline scripts/setup-fix.bat]: what's the reason to do the 'cp -R' above?

File: scripts/setup-fix.bat [hunk 2]
```diff
@@ -0,0 +1,11 @@
+if ! [ -x "$(command -v python)" ];
+then 
+	echo Python environment notfound..
```
Comment (yliaog) [inline scripts/setup-fix.bat]: probably add double quotes around the echo string

File: scripts/setup-fix.bat [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+else 
+	echo Python environment found..
```
Comment (yliaog) [inline scripts/setup-fix.bat]: ditto

File: scripts/setup-fix.bat (hunk 1 above)
Comment (Sai-Adarsh) [reply scripts/setup-fix.bat]: > what's the reason to do the 'cp -R' above?

"When we clone submodule with git the folders in the /kubernetes/base is supposed to be symbolically linked in the /kubernetes directory. In Windows this is not done. For a Windows developer, we’ll need to have a script copying all the base/* folders into */. "

^ is what @micw523 requested in his issue. so 'cp -R' does the job.

File: scripts/setup-fix.bat (hunk 3 above)
Comment (Sai-Adarsh) [reply scripts/setup-fix.bat]: > ditto

cool. will do the necessary changes

**Comment by micw523** [comment]
To be honest this looks like bash instead of a Windows bat. You could probably use it in a Cygwin but I think Windows developers would prefer a genuine bat.

**Comment by Sai-Adarsh** [comment]
@micw523 done with the changes. could you check on the new commit :)

**Comment by Sai-Adarsh** [comment]
> I have a few comments -
> 
> * Could you break your command into several lines? It is difficult to decipher at its current state
> * xcopy is considered an external program and has been deprecated since '07. Could you try using copy or robocopy?

done with the changes.

File: scripts/setup-fix.bat [hunk 4]
```diff
@@ -0,0 +1,5 @@
+( 	where python >null
+	echo Python environment found..
```
Comment (micw523) [inline scripts/setup-fix.bat]: In this case Python environment found will be printed regardless.

Comment (micw523) [reply scripts/setup-fix.bat]: ```python > null``` and ```found.```

Comment (Sai-Adarsh) [reply scripts/setup-fix.bat]: > `python > null` and `found.`

cool. done with the changes.

**Comment by Sai-Adarsh** [comment]
> Please fix your indents - the conditional statements are not aligned.
> Also, one period for output message is sufficient.
> On the last line `../` - the slash is not needed and on Windows you're supposed to use backslash anyway so `cd ..` should be good enough

done.

File: scripts/setup-fix.bat (hunk 3 above)
Comment (micw523) [reply scripts/setup-fix.bat]:
```suggestion
	echo Python environment found.
```

File: scripts/setup-fix.bat [hunk 5]
```diff
@@ -0,0 +1,6 @@
+(	python --version>nul 2>&1 && (
+	echo Python environment found.
```
Comment (roycaihw) [inline scripts/setup-fix.bat]: just out of curiosity: was the double quote change @yliaog  suggested reverted for some reason?

Comment (Sai-Adarsh) [reply scripts/setup-fix.bat]: > 
> 
> just out of curiosity: was the double quote change @yliaog suggested reverted for some reason?

works fine without double qoute, so thought of reverting the change :)

**Comment by Sai-Adarsh** [comment]
> 
> 
> /hold
> 
> Please add comments in the bat describing what it's for. Probably also name the file "windows-setup-fix.bat" to make it more clear. LGTM otherwise

cool.

**Comment by Sai-Adarsh** [comment]
> 
> 
> /hold
> 
> Please add comments in the bat describing what it's for. Probably also name the file "windows-setup-fix.bat" to make it more clear. LGTM otherwise

@roycaihw could you check on the new changes?

**Comment by roycaihw** [comment]
Could you add the comments to the beginning of the file? Basically I was asking for a description so that people can tell what it's for and when it should be used. I think we don't need inline comments.

Something like "Install python client in Windows. Windows doesn't have symbolic link that this repo uses. This batch script provides a workaround for symbolic link by copying the folders over."

Also please squash your commits before submit

/hold cancel

**Comment by micw523** [comment]
Will lg()tm after squash

**Comment by Sai-Adarsh** [comment]
> 
> 
> Will lg()tm after squash

cool.

**Pull request closed without merging**